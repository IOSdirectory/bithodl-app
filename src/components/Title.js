import React, { Component } from 'react';
import { StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import StyledText from './StyledText';

const styles = StyleSheet.create({
  title: {
    fontSize: 18,
    fontWeight: '600',
    letterSpacing: 1,
    marginBottom: 10,
    textAlign: 'center'
  }
});

export default class Title extends Component {
  render() {
    return (
      <StyledText style={[styles.title, this.props.style]}>
        {this.props.children}
      </StyledText>
    );
  }
}

Title.propTypes = {
  style: PropTypes.any,
  children: PropTypes.node
};
