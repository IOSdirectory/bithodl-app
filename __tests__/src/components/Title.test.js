import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import Title from '../../../src/components/Title';

describe('Title', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Title>
        <Text>32a29072-5334-4ad1-a86d-37f6c2639449</Text>
      </Title>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
