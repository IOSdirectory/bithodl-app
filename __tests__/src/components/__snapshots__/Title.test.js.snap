// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  numberOfLines={undefined}
  style={
    Array [
      Object {
        "color": "#504F63",
        "fontFamily": "Avenir Next",
        "fontWeight": "400",
      },
      Array [
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
          "letterSpacing": 1,
          "marginBottom": 10,
          "textAlign": "center",
        },
        undefined,
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    32a29072-5334-4ad1-a86d-37f6c2639449
  </Text>
</Text>
`;
